### PR TITLE
Enable benchmark summary for release

### PR DESCRIPTION
**Pull request by XSAM**
It enables the job summary for triggered PRs. (it doesn't create a comment in PRs) Example: https://github.com/benchmark-action/github-action-benchmark/issues/159#issuecomment-1871746703

### CODE REVIEW

**Review by dmathieu — APPROVED** [review]

**Review by dashpole — APPROVED** [review]

**Comment by pellared** [comment]
How is it supposed to work?

Notice

```
on:
  push:
    tags:
      - v1.* 
  workflow_dispatch:
```

**Comment by XSAM** [comment]
> > How is it supposed to be triggered?
> > Notice
> > ```
> > on:
> >   push:
> >     tags:
> >       - v1.* 
> >   workflow_dispatch:
> > ```
> 
> IIUC, when we make a tag it triggers a job to run. I was thinking that maybe triggering this job when we create the release PR could be better.

But this can be replaced by running benchmarks against every commit in #4537

I think it is ok for this PR to just enable the summary feature.

**Comment by pellared** [comment]
> I think it is ok for this PR to just enable the summary feature.

Can you please showcase what would be the output on a fork?

**Comment by XSAM** [comment]
> > I think it is ok for this PR to just enable the summary feature.
> 
> Can you please showcase what would be the output on a fork?

I cannot trigger the benchmark CI on this PR, but here is the result from other projects: https://github.com/benchmark-action/github-action-benchmark/issues/159#issuecomment-1871746703

**Comment by pellared** [comment]
> I cannot trigger the benchmark CI on this PR, 

Cannot you do it on your fork? https://github.com/XSAM/opentelemetry-go/tree/summary-for-benchmark

**Comment by XSAM** [comment]
> > I cannot trigger the benchmark CI on this PR,
> 
> Cannot you do it on your fork? https://github.com/XSAM/opentelemetry-go/tree/summary-for-benchmark

The summary looks like this https://github.com/XSAM/opentelemetry-go/actions/runs/9786770170/attempts/1#summary-27022201692

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -37,3 +37,5 @@ jobs:
           gh-pages-branch: benchmarks
           fail-on-alert: true
           alert-threshold: "400%"
+          # Enable Job Summary for PRs
```
Comment (pellared) [inline .github/workflows/benchmark.yml:40]:
```suggestion
          # Add benchmark summary to GitHub workflow run report
```

**Review by pellared — APPROVED** [review]

**Pull request merged**